app/controllers/simulations_controller.rb

```diff
@@ -201,6 +201,12 @@ def mark_as_complete
             @experiment.progress_bar_update(@simulation_run.index, 'error')
           else
             @experiment.progress_bar_update(@simulation_run.index, 'done')
+            # Increment finished simulation counter in simulation manager record
+            facade = InfrastructureFacadeFactory.get_facade_for(@simulation_run.infrastructure)
+            simulation_manager_record = facade.get_sm_record_by_id(@simulation_run.sm_uuid)
+            simulation_manager_record.finished_simulations = 0 unless simulation_manager_record.finished_simulations
```

Comment:
Nie wiem czemu mi to nie przyszło do głowy wcześniej, ale powinieneś zrobić lazy loading:
simulation_manager_record.finished_simulations ||= 0